crates/client/block_import/src/verify_apply.rs

```diff
@@ -160,7 +162,7 @@ fn check_parent_hash_and_num(
     };
 
     let block_number = if let Some(block_n) = unverified_block_number {
-        if block_n != expected_block_number {
+        if block_n != expected_block_number && !validation.ignore_block_order {
```

Comment:
yes if for x reason the block order is disrupted then it will throw an error. If unswafe-starting-block is set then we assume you dont want your state to be ordered